zerver/lib/streams.py

```diff
@@ -1046,6 +1046,9 @@ def get_setting_values_for_group_settings(
         else:
             anonymous_group_ids.append(group.id)
 
+    if len(anonymous_group_ids) == 0:
```

Comment:
I checked and there are no DB queries when `anonymous_group_ids` so we can also skip adding this.

Reply:
Probably clearer to have it.